src/components/Badge/index.tsx

```diff
@@ -0,0 +1,27 @@
+import classNames from 'classnames';
+import React, { FC } from 'react';
+
+import styles from './Badge.module.css';
+
+import { IBadgeProps } from './IBadge.d';
+
+const Badge: FC<IBadgeProps> = ({
+  children = '',
+  color = 'primary',
+  className,
+}) => {
+  const BadgeClasses = classNames(
+    `${styles['shared']} ${styles[`variant-${color}`]}`,
+    {
+      [styles['empty-state']]: !children,
+      [styles['filled-state']]: Boolean(children),
+    },
+    className
+  );
+  const capitalizedFirstLetter = children?.charAt(0).toUpperCase();
+  const label = `${capitalizedFirstLetter}${children?.slice(1)}`;
+
+  return <label className={BadgeClasses}>{label}</label>;
```

Comment:
I think using the label tag is not semantically correct for this case. Usually it is associated with an input.

Reply:
this component is only responsible for previewing content, there won't be any sort of interactions the user can do for this one,
so I guess `label` tag usage is fine in this case

Reply:
I mean as there are  not any interactions so we do not need a label. As per [MdN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label)   and  [W3C](https://www.w3.org/TR/WCAG20-TECHS/H44.html). 

> Using label elements to associate text labels with form controls 

I suggest  to replace it with `span`